[US-0005] - Admin View Pending Creator Requests

Aw1221 opened the pull request:
## Description

<!-- Descreva claramente a mudança implementada neste Pull Request -->

- Related to: [Issue #39]
## Objective

<!-- Explique o motivo dessa alteração e os problemas que ela resolve -->

## Acceptance Criteria

- [x] Admin can view a list of all pending creator requests.
- [x] Each request displays creator name, email, and submission date.
- [x] Admin can open a detailed view of each request.
- [x] Admin can approve or reject a request from the same interface.
- [x] Status updates reflect immediately after an action is taken.
- [x] The list automatically updates when a request changes status.

## Related User Story

As an Admin,
I want to see all pending creator requests,
so that I can review and approve or reject them efficiently.



## Screenshot 

<img width="1455" height="691" alt="image" src="https://github.com/user-attachments/assets/7ccede9c-f44b-4270-ba18-87d0acc80569" />

<img width="626" height="643" alt="image" src="https://github.com/user-attachments/assets/3f05dc94-cf64-4428-9f70-b33bcd5cec3c" />

<img width="1123" height="430" alt="image" src="https://github.com/user-attachments/assets/1f4badd3-218d-45c0-ab41-4e1af414b6ec" />

<img width="1445" height="693" alt="image" src="https://github.com/user-attachments/assets/520b5d07-bb4c-4867-b3b5-3d8e72f039d5" />

<img width="1133" height="577" alt="image" src="https://github.com/user-attachments/assets/ce80d6fd-fc7a-451a-b9a8-ed622c8f9457" />

<img width="1623" height="693" alt="image" src="https://github.com/user-attachments/assets/e81e59fd-2121-4950-a933-5a6f9c502516" />

<img width="1509" height="693" alt="image" src="https://github.com/user-attachments/assets/acfb1c9e-185a-4102-adf5-f9764aec4a3e" />

## Important

To test this PBI you need to set the variable isAdmin to true on your content creator instance

## What has changed:

### Frontend
Users table improvements
Displays:
First / Last name
Email
Status (pending, approved, rejected)
Role (admin or content creator)
Status now updates immediately after performing approve/reject.


Individual content creator view
Fetches data from the new endpoint /api/user-info/:id.
Shows the content creators:
Personal infomation
Academic experience
Work experience
Buttons for:
Approve, which calls endpoint with APPROVED statusValue
Reject, which calls endpoint with  REJECTED statusValue + rejectionReason

Miscellaneous
Made sure to use i18n for everything so its readable in english and portuguese 
Changed error messages to better show what is happening


### Backend
New custom endpoint: /api/user-info
Introduced a custom controller with:
find(): returns list of all content creators 
findOne(id):  returns individual details for a single applicant.

Added isAdmin support
Extended content-creator to include isAdmin: boolean. This means you can now be an admin on educado and get access to admin only features. 

New status update endpoint
Created a custom route + controller, which is responsible for updating the status of an content creator
PATCH /api/content-creators/:id/status. Decided to go with patch as i only need to update 1 single attribute and dont want to send the whole object back everytime we update

These are the most important changes in this PR

Aw1221 committed:
Addint all possible status filters and defaulting to pending

Aw1221 committed:
Applying the filter to take the new status into consideration

# Conflicts:
#	web/src/unplaced/Admin/Users/UsersTableAdmin.tsx

Aw1221 committed:
Adding Buttons for every status

Aw1221 committed:
Creating admin boolean value, and validation that a user is an admin

# Conflicts:
#	openapi/strapi-spec.json
#	strapi/src/extensions/documentation/documentation/1.0.0/full_documentation.json

# Conflicts:
#	strapi/src/extensions/documentation/documentation/1.0.0/full_documentation.json

Aw1221 committed:
Adding logic for adminds so u can retrieve,delete,change role, get the list of admins and their record

Aw1221 committed:
Fixing controller, adding statusValues to content creator

Aw1221 committed:
Admin works on web

 Conflicts:
	openapi/strapi-spec.json
	strapi/public/openapi/strapi-spec.json
	strapi/src/api/content-creator/controllers/content-creator.ts
	strapi/src/extensions/documentation/documentation/1.0.0/full_documentation.json

Aw1221 committed:
Removing policy for testing, simplying admin policy

Aw1221 committed:
Admin panel displays content creators now

Aw1221 committed:
You can now view a specific content creator

Aw1221 committed:
Now able to approve and reject content creators

Aw1221 committed:
Admins can now approve or reject content creators

Aw1221 committed:
Made translation so admin panel can be read in both portuguese and english

Aw1221 committed:
Finalised last translation issues

Aw1221 committed:
Commenting out test as they dont run right now

Aw1221 committed:
Fixed github actions issue

PeterDunlop commented on web/src/unplaced/Admin/UserInfo/UserGenericContainerComponent.tsx:
Please address the Sonarqube error here. Is AuthServices and other stuff actually asynchronous? If  not please remove the await.

PeterDunlop reviewed (changes requested):
Everything looks pretty good, but there's a strange thing with asynch functions in one of the files. Please address this otherwise it's fine.

Aw1221 committed:
Making sure acceptApplication and rejectApplication are actually asynchronous

PeterDunlop reviewed (dismissed):
Looks fine. Last remaining sonarqube warnings are not vital and Admin Panel is a huge requirement to do a lot of features and must be included.

CubeKR1 commented on strapi/src/api/content-creator/controllers/content-creator.ts:
```diff
@@ -75,35 +75,87 @@ export default factories.createCoreController('api::content-creator.content-crea
 
 
-
-            const jwtContentCreator: ContentCreator = {
```
Earlier in the project, we defined a type for JWTs of Content Creators and Students. This is in ./strapi/src/helpers/types.ts. It would be nice for maintainability if this type was extended with the new attribute (isAdmin), so that we can keep JWT types consistent across the project.

Aw1221 replied:
Thank you i could not find this data type even though i searched throughout the project. Makes alot more sense to update it to include isAdmin rather than create a whole new one

CubeKR1 commented on strapi/src/api/content-creator/controllers/content-creator.ts:
```diff
@@ -75,35 +75,87 @@ export default factories.createCoreController('api::content-creator.content-crea
-                },
-            });
+                verifiedAt: user.verifiedAt ? new Date(user.verifiedAt).toISOString() : null,
+                isAdmin: !!user.isAdmin,
```
Shouldn't it just be user.isAdmin since we know it is a boolean.

CubeKR1 commented on strapi/src/api/content-creator/controllers/content-creator.ts:
```diff
@@ -75,35 +75,87 @@ export default factories.createCoreController('api::content-creator.content-crea
+        }
+
+        if (statusValue === "REJECTED") {
+            (data as any).rejectionReason = rejectionReason ?? null;
```
Should be able to remove as any because data has a defined type above.

CubeKR1 commented on strapi/src/api/content-creator/controllers/content-creator.ts:
```diff
@@ -75,35 +75,87 @@ export default factories.createCoreController('api::content-creator.content-crea
+            await strapi.documents("api::content-creator.content-creator")
+               .update({
+                   documentId: id,
+                   data,
```
My Content Creator collection type does not have a rejectionReason in Strapi, so this attribute is not stored in the content creator, even though it is sent as a part of data if the content creator is rejected. Am I missing something, or what is supposed to happen?

Aw1221 replied:
It is now implemented and saves the rejectionReason

CubeKR1 commented on strapi/src/api/users/controllers/users.ts:
I can see that the these controllers and the ones in user-info relate to the content creator, but the naming is ambiguous. Might need some better naming to ensure that these controllers we do not mistake them for being related to students, who are also users.

CubeKR1 replied:
That makes sense. I think that is a good argument for why they should be separated.

CubeKR1 reviewed (changes requested):
I primarily looked at the backend and left some comments that can hopefully improve the code a bit. I also ran the program, and it works nicely (only the institution page under the admin panel keeps returning errors (might need to be mocked instead before the functionality is added)). There is still some work needed on the styling, but that can be future work.

Aw1221 committed:
Handled PR comments

# Conflicts:
#	strapi/src/extensions/documentation/documentation/1.0.0/full_documentation.json

CubeKR1 reviewed (dismissed):
Looks good now. Great work!

Aw1221 committed:
Fixed translation issues

hjorth30 reviewed (approved):
From what I understand the files in /web/src/unplaced are all unused, but you have changed some in it. Are they used and if so can they be moved to features or shared?
Other than that it all looks good. Well done!

Aw1221 commented:
CO-CONTRIBUTERS: @Aw1221
